commit 4c581e8bdbdc4b7ba6647815d446e8f41f01a9c0 (refs/changes/68/1468/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-07T14:47:29-08:00 (2 years, 8 months ago)
    
    joyent/node-mname#16 broken on node 0.10 due to use of Map

diff --git a/lib/dns-buffer.js b/lib/dns-buffer.js
index 34a9620..d26b7ae 100644
--- a/lib/dns-buffer.js
+++ b/lib/dns-buffer.js
@@ -311,6 +311,22 @@ if (Buffer.prototype.readUIntBE !== undefined &&
 
 }
 
+/*
+ * We are expected to run on node v0.10, which lacks Map. Polyfill just the
+ * bits of it we actually need here. Map is a bit faster than an object for our
+ * task, but an object will do as a substitute on 0.10.
+ */
+if (typeof (Map) !== 'function') {
+        var Map = function () {
+                this._obj = {};
+        };
+        Map.prototype.set = function (k, v) {
+                this._obj[k] = v;
+        };
+        Map.prototype.get = function (k) {
+                return (this._obj[k]);
+        };
+}
 
 function NameOffsetCache() {
         this.root = new Map();
diff --git a/package.json b/package.json
index 2d4caba..ecf1eed 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mname",
   "description": "DNS server library for node.js",
-  "version": "1.3.3",
+  "version": "1.3.4",
   "author": "arekinath <alex@cooperi.net>",
   "contributors": [
     "Mark Cavage",
